src/components/sections/Hero.section.tsx

```diff
@@ -1,8 +1,65 @@
-const HeroSection = () => {
+import React, { useState, useEffect } from 'react';
+import { HeroIllustration } from '@assets';
+import {
+  LinkedInIcon,
+  TwitterIcon,
+  InstagramIcon,
+  DiscordIcon
+} from '@assets';
+
+
+const HeroSection: React.FC = () => {
+  const [countdown, setCountdown] = useState<string>('');
+
+  // Function to calculate the countdown
+  const calculateCountdown = () => {
+    const targetDate = new Date('2024-04-06T00:00:00'); // Set your target date here
+    const currentDate = new Date();
+    const difference = targetDate.getTime() - currentDate.getTime();
+
+    const days = Math.floor(difference / (1000 * 60 * 60 * 24));
+    const hours = Math.floor((difference % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+    const minutes = Math.floor((difference % (1000 * 60 * 60)) / (1000 * 60));
+    const seconds = Math.floor((difference % (1000 * 60)) / 1000);
+
+    setCountdown(`${days} ${hours} ${minutes} ${seconds}`);
+  };
+
+  // Update the countdown every second
+  useEffect(() => {
+    const interval = setInterval(() => {
+      calculateCountdown();
```

Comment:
I might be wrong here, but doesn't this function needs to be added to the dependency list in `useEffect`? The original function will be refined, but useEffect will not run again. I guess it works if the function only uses a state setter which remains the same across renders, but I don't think this is the ideal way to handle it?

I would prefer moving all the logic of the function in the callback for `setInterval` itself.

@aidantrabs what are your thoughts on this?

Reply:
Yeah I believe youre right @juancwu 

@norachams Here is what it should look like - 
```js
import React, { useState, useEffect, useCallback } from 'react';
// rest of ya imports

const HeroSection: React.FC = () => {

  const calculateCountdown = useCallback(() => { // put da callback here
    // all da function content
  }, []); // Make an empty dependency array here since it has no dependencies here

  useEffect(() => {
    const interval = setInterval(() => {
      calculateCountdown();
    }, 1000);
    return () => clearInterval(interval);
  }, [calculateCountdown]); // Add the calc countdown function here 

  return (
      // blah blah blah
  );
};

export { HeroSection };
```

Reply:
Fixed it and pushed the new code.